c++/c++-quiz.md

```diff
@@ -0,0 +1,272 @@
+
+    - [] typedef struct {
+    		int sunday:1;
+    		int monday:1;
+    		// more days
+    		int friday:1;
+    		int satarday:1;
+    		- [x]ekdays;  That syntax says that each variable size is 1 bit. 'bit' is not a type in C++.
+    - [] typedef char[7]: weekdays;
+    - [] typedef struct {
+    		bit sunday:1;
+    		bit monday:1;
+    		// more days
+    		bit fridyas:1;
+    		bit satarday:1;
+    		} weekdays;
+    - [] typedef struct {
+    		bit sunday;
+    		bit monday;
+    		// more days
+    		bit friday;
+    		bit satarday;
+    		} weekdays;
+
+#### Q12. What is an lvalue?
+- []re>a) It's a constant expression , meaning an expression composed of constants and operations.
+- [x] It's an expression that represents an object with an address.
+- [] It's an expression suitable for the left hand side operand in binary operation.
+- [] It's a location value , meaning a memory address suitable for assigning to pointer or reference.
+</pre>
+#### Q13. What does auto type specifier do in this line of code (since c++ 11)?
+
+    - [] It specifies that the type of x will be deducted from the initializer in this case,double.
+    - [] It specifies that the type of x is automatic meaning that if can be assigned different type of data throughout the program.
+    - [x] It specifies that x is a variable with automatic storage duration.
+    - [] It soecifies that more memory will be allocated for x in case it needs more space, avoiding loss of data due to overflow.
+
+#### Q14. What is a class template?
+
+    - [x] It's a class written with the generic programming , specifying behavior in terms of type parameter rather than specific type.
+    - [] It's a blank superclass intended for inheritence and polymorphism.
+    - [] It's class that only consists of member variable , with no constructor , destructor nor member functions.
+    - [] It's skelton source code for a class where the programming has to fill in specific parts to define the data types and algorithms used.
+
+#### Q15. What is the ternary operator equivalent to this code snippet?
+
+    if(x)
+    y=a;
+    else
+    y=b;
+
+    - [] y=a?b:x;
+    - [] y=if(x?a:- [];
+    - [] y=(x&a)?a:(x&b)?b:0;
+    - [x] y=x?a:b;
+
+#### Q16. What is the output of this code? What is the output of this code?
+
+    #include <iostream>
+    int main()
+    {
+    	int x=10, y=20;
+    	std::cout << "x = " << x++ << " and y = " << --y << std::endl;
+    	std::cout << "x = " << x-- << " and y = " << ++y << std::endl;
+    	return(0);
+    }
+
+    - [] x = 10 and y = 20 x = 11 and y = 19
+    - [] x = 11 and y = 19 x = 10 and y = 20
+    - [x] x = 10 and y = 19 x = 11 and y = 20
+    - [] x = 11 and y = 20 x = 10 and y = 19
+
+#### Q.17 What is the meaning of the two parts specified between parentheses in a range-based for loop, separated by a colon?What is the meaning of the two parts specified between parentheses in a range-based for loop, separated by a colon?
```

Comment:
```suggestion
#### Q17. What is the meaning of the two parts specified between parentheses in a range-based for loop, separated by a colon?What is the meaning of the two parts specified between parentheses in a range-based for loop, separated by a colon?
```

Reply:
no problem, I will do it: Instead of Q.17, I do Q17.

Reply:
sorry, I just noticed that it was not originally, but would be great if you can improve it! Thank you in advance, Timo!